sycl/test/atomic_ref/accessor.cpp

```diff
@@ -0,0 +1,59 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <numeric>
+#include <vector>
+using namespace sycl;
+using namespace sycl::intel;
+
+// Equivalent to add_test from add.cpp
+// Uses atomic_accessor instead of atomic_ref
+template <typename T>
+void accessor_test(queue q, size_t N) {
+  T sum = 0;
+  std::vector<T> output(N, 0);
+  {
+    buffer<T> sum_buf(&sum, 1);
+    buffer<T> output_buf(output.data(), output.size());
+
+    q.submit([&](handler &cgh) {
+#if __cplusplus > 201402L
+      auto sum = atomic_accessor(sum_buf, cgh, relaxed_order, device_scope);
+      static_assert(std::is_same<decltype(sum), atomic_accessor<T, 1, intel::memory_order::relaxed, intel::memory_scope::device>>::value, "atomic_accessor type incorrectly deduced");
+#else
+      auto sum = atomic_accessor<T, 1, intel::memory_order::relaxed, intel::memory_scope::device>(sum_buf, cgh);
```

Comment:
Minor. I guess this branch is not compiled -> not tested.

Reply:
Fixed in df7fddd79b7ca8c348b04affa99810b8cdfe206b.